src/ppom-controller.ts

```diff
@@ -1014,15 +1056,17 @@ export class PPOMController extends BaseControllerV2<
       // Invoke provider and return result
       this.#provider.sendAsync(
         createPayload(method, params),
-        (error: Error, res: any) => {
-          if (error) {
+        (error, res: JsonRpcSuccess<Json>) => {
+          if (!error) {
+            resolve(res);
+          } else if (isJsonRpcError(error)) {
             resolve({
               jsonrpc: '2.0',
               id: IdGenerator(),
               error,
             });
           } else {
-            resolve(res);
+            reject(error);
           }
```

Comment:
We could handle this differently (e.g. `console.error`) or simply drop this branch entirely.